Bottom: b70b288cb02fd8e77c80f06128871e14eb4b4ec3
Top:    b3cc86db60f77cc9ad41b27aa77f4c90fced67eb
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2010-03-18 20:35:52 +0000

Refresh of fix-poly_contourincontour-test

---

diff --git a/src/polygon1.c b/src/polygon1.c
index 308519a..0f7bf98 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -2969,18 +2969,22 @@ poly_ContourInContour (PLINE * poly, PLINE * inner)
   VNODE *pt;
   assert (poly != NULL);
   assert (inner != NULL);
-  if (cntrbox_inside (inner, poly)) {
+  if (cntrbox_inside (inner, poly))
 #if 1
-    pt = &inner->head;
-    /* HACK: Check all points on the contour being tested, because we don't
-     *       want to falsely return that two contours are inside each other
-     *       if they just touch at a few points.
-     */
-    do {
-      if (!poly_InsideContour (poly, pt->point))
-        return 0;
-    } while ((pt = pt->next) != &inner->head);
-    return 1;
+    { /* FIXME: This is SLOW!!
+       *
+       * Check all points on the contour being tested, because we don't
+       * want to falsely return that two contours are inside each other
+       * if they just touch at a few points.
+       */
+      pt = &inner->head;
+      do
+        {
+          if (!poly_InsideContour (poly, pt->point))
+            return 0;
+        } while ((pt = pt->next) != &inner->head);
+      return 1;
+    }
 #else
     return poly_InsideContour (poly, inner->head.point);
 #endif
